inst/stan/data/observation_model.stan

```diff
@@ -3,17 +3,22 @@
   real phi_mean;                      // Mean and sd of the normal prior for the
   real phi_sd;                        // reporting process
   int week_effect;                   // length of week effect
-  int truncation;                    // 1/0 indicating if truncation should be adjusted for
-  real trunc_mean_mean[truncation];  // truncation mean of mean
-  real trunc_mean_sd[truncation];    // truncation sd of mean
-  real trunc_sd_mean[truncation];    // truncation mean of sd
-  real trunc_sd_sd[truncation];      // truncation sd of sd
-  int<lower = 1> trunc_max[truncation];    // maximum truncation supported
-  int trunc_fixed[truncation];    // whether the truncation distribution is fixed
-  int trunc_dist[truncation];    // 0 = lognormal; 1 = gamma
   int obs_scale;                     // logical controlling scaling of observations
   real obs_scale_mean;               // mean scaling factor for observations
   real obs_scale_sd;                 // standard deviation of observation scaling
   real obs_weight;                   // weight given to observation in log density
   int likelihood;                    // Should the likelihood be included in the model
   int return_likelihood; // Should the likehood be returned by the model
+
+  int<lower = 0> trunc_n;                  // number of trunc distribution distributions
+  int<lower = 0> trunc_n_p;                // number of parametric trunc distributions
+  int<lower = 0> trunc_n_np;                // number of nonparametric trunc distributions
```

Comment:
Noting that this would also have to be changed to support multiple observations (see #313) and either be specified as a nested list or given to the observation model. 

A much bigger potential redesign would be to make delays and truncations be nested in each observation model. This could have quite a few benefits but would come at the cost of needing quite a bit of redesign work.

Reply:
This now has a single delay vector as a first step towards this.